pkg/pillar/cmd/zedagent/handlelocation.go

```diff
@@ -148,15 +151,16 @@ func publishLocationToController(locInfo *info.ZInfoLocation, iteration int) {
 
 	const bailOnHTTPErr = false
 	const withNetTrace = false
-	ctxWork, cancel := zedcloud.GetContextForAllIntfFunctions(zedcloudCtx)
-	defer cancel()
-	rv, err := zedcloud.SendOnAllIntf(ctxWork, zedcloudCtx, infoURL,
-		size, buf, iteration, bailOnHTTPErr, withNetTrace)
-	if err != nil {
-		// Hopefully next timeout will be more successful
-		log.Errorf("publishLocationToController: failed (status %d): %v", rv.Status, err)
-		return
-	}
+	key := "location:" + devUUID.String()
+
+	// Ignore errors for all destinations. Even for the controller
+	// we can't stall the queue on error, because this call is
+	// recurring.
+	ignoreErr := dest
+	items := createDeferredItemsByDest(ctx, dest, ignoreErr, key, buf, size,
+		bailOnHTTPErr, withNetTrace, info.ZInfoTypes_ZiLocation)
+	zedcloud.AddDeferredList(zedcloudCtx, items)
+	zedcloud.HandleDeferred(zedcloudCtx, time.Now(), 0, false)
```

Comment:
I think this should at most send len(items) thus instead of a sendOne bool we need to make HandleDeferred that a sendCount int, which you set to len(items) across the board.
Basically we only try to send as many items as were added by this function.